bin/tx-prover/src/error.rs

```diff
@@ -0,0 +1,23 @@
+use axum::http::uri::InvalidUri;
+use thiserror::Error;
+
+// TX PROVER ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum TxProverProxyError {
+    #[error("invalid uri error")]
+    InvalidURI(#[source] InvalidUri),
+    #[error("failed to connect to worker")]
+    ConnectionFailed(#[source] tonic::transport::Error),
+    #[error("failed to create backend for worker")]
+    BackendCreationFailed(#[source] Box<pingora::Error>),
+    #[error("app logic not found")]
+    AppLogicNotFound,
```

Comment:
Could we add a bit more info about this error? From the name and message it is not really clear to me what this error is about.